debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/J9ConstantPoolCommand.java

```diff
@@ -0,0 +1,539 @@
+					if (i < ramCPCount) {
+						J9ObjectPointer stringObject = J9RAMStringRefPointer.cast(ramCPP).stringObject();
+						if (stringObject.notNull()) {
+							ramInfo = "!j9object " + stringObject.getHexAddress();
+						} else {
+							ramInfo = "(unresolved)";
+						}
+					}
+					value = J9UTF8Helper.stringValue(J9ROMStringRefPointer.cast(romCPP).utf8Data());
+				} else if (cpType == J9ConstantPool.J9CPTYPE_CONSTANT_DYNAMIC) {
+					try {
+						if (i < ramCPCount) {
+							J9RAMConstantDynamicRefPointer ref = J9RAMConstantDynamicRefPointer.cast(ramCPP);
+							if (ref.value().notNull()) { // resolved
+								ramInfo = "!j9object " + ref.value().getHexAddress() + " (resolved)";
+							} else {
+								J9ObjectPointer exception = ref.exception();
+								if (exception.notNull()) {
+									J9ClassPointer clazz = J9ObjectHelper.clazz(exception);
+									J9ClassPointer throwable = J9VMConstantPoolClassRefAt(vm, J9VMCONSTANTPOOL_JAVALANGTHROWABLE).value();
+									if (exception.getAddress() == vm.voidReflectClass().classObject().getAddress()) {
+										ramInfo = "NULL (resolved)";
+									} else if (J9ClassHelper.isSameOrSuperClassOf(throwable, clazz)) { // resolved exception
+										ramInfo = "!j9object " + exception.getHexAddress() + " (exception)";
+									} else { // resolving by a thread object
+										ramInfo = "!j9object " + exception.getHexAddress() + " (resolving Thread)";
+									}
+								} else {
+									ramInfo = "(unresolved)";
+								}
+							}
+						}
+
+						J9ROMConstantDynamicRefPointer romRef = J9ROMConstantDynamicRefPointer.cast(romCPP);
+						SelfRelativePointer callSiteData = SelfRelativePointer.cast(romClass.callSiteData());
+						long callSiteCount = romClass.callSiteCount().longValue();
+						U16Pointer bsmIndicies = U16Pointer.cast(callSiteData.addOffset(4 * callSiteCount));
+						U16Pointer bsmData = bsmIndicies.add(callSiteCount);
+						long bsmIndex = romRef.bsmIndexAndCpType().longValue() >> J9DescriptionCpTypeShift & J9DescriptionCpBsmIndexMask;
+
+						long argCount = 0;
+						for (long j = 0; j < bsmIndex; j++) {
+							argCount = bsmData.add(1).at(0).longValue();
+							bsmData = bsmData.add(argCount + 2);
+						}
+						argCount = bsmData.add(1).at(0).longValue();
+						long bsmMethodRefIndex = bsmData.at(0).longValue();
+						romInfo = "#" + bsmMethodRefIndex;
+
+						J9ROMNameAndSignaturePointer nameAndSig = romRef.nameAndSignature();
+						value = String.format("%s:%s #%d(",
+											J9UTF8Helper.stringValue(nameAndSig.name()),
+											J9UTF8Helper.stringValue(nameAndSig.signature()),
+											bsmMethodRefIndex);
+						for (long j = 1; j <= argCount; j++) {
+							value += "#" + bsmData.add(1 + j).at(0).longValue();
+							if (j < argCount) {
+								value += ", ";
+							}
+						}
+						value += ")";
+					} catch (NoSuchFieldException | NoClassDefFoundError e) {
+						throw new CorruptDataException(e);
```

Comment:
Please list exceptions alphabetically and add a comment explaining why it's reasonable to classify it as corruption.